editor/src/document/movement_handler.rs

```diff
@@ -17,16 +17,27 @@ use std::collections::VecDeque;
 #[impl_message(Message, DocumentMessage, Movement)]
 #[derive(PartialEq, Clone, Debug, Serialize, Deserialize)]
 pub enum MovementMessage {
-	MouseMove { snap_angle: Key },
+	MouseMove {
+		snap_angle: Key,
+		wait_for_snap_angle_release: bool,
+		snap_zoom: Key,
+		zoom_from_viewport: Option<DVec2>,
```

Comment:
When reviewing the code, I've been rather stumped by this variable name. Can you think of any clearer name for it? Or if you want to describe for me what it does, I can perhaps suggest a name.

Reply:
It is an optional viewport position around which to zoom (defaults to the centre).

Reply:
zoom_viewport_pivot?